framework/transport/ieee1905_transport/ieee1905_transport_broker.cpp

```diff
@@ -0,0 +1,368 @@
+
+#include <easylogging++.h>
+
+namespace beerocks {
+namespace transport {
+namespace broker {
+
+//////////////////////////////////////////////////////////////////////////////
+////////////////////////// Local Module Definitions //////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+// Number of concurrent connections on the server socket
+static constexpr int listen_buffer_size = 10;
+
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////// Local Module Functions ///////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+static bool is_restricted_type(uint32_t type)
+{
+    // TODO: Until VS message filtering is implemented (based on action and action_op)
+    //       assume there are no restricted types. After the implementation,
+    //       subscribing to BROKER specific message should be restricted.
+
+    return false;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////// Broker Implementation ////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+BrokerServer::BrokerServer(const std::string &broker_uds_path, SocketEventLoop::TimeoutType timeout)
+    : SocketEventLoop(timeout)
+{
+    m_broker_uds_path = broker_uds_path;
+
+    // Server sockets event handlers
+    EventHandlers server_socket_handlers{
+        // Accept incoming connections
+        .on_read =
+            [&](EventType socket, EventLoopType &loop) {
+                // cppcheck falsely detects the call to socket_connected() as a call to a
+                // virtual method from whithin the constructor, so suppress this warning,
+                // by explicitly calling BrokerServer::socket_connected().
+
+                // Handle connections to one of the server sockets
+                if (!BrokerServer::socket_connected(
+                        std::dynamic_pointer_cast<SocketServer>(socket))) {
+                    // NOTE: Do NOT stop the broker on connection errors...
+                }
+
+                return true;
+            },
+
+        // Not implemented
+        .on_write   = nullptr,
+        .on_timeout = nullptr,
+
+        // Fail on server socket disconnections or errors
+        .on_disconnect =
+            [&](EventType socket, EventLoopType &loop) {
+                LOG(ERROR) << "Broker socket disconnected!";
+                return false;
+            },
+        .on_error =
+            [&](EventType socket, EventLoopType &loop) {
+                LOG(ERROR) << "Broker socket error!";
+                return false;
+            },
+    };
+
+    // Connect to the message broker
+    m_broker_socket_uds = std::make_shared<SocketServer>(m_broker_uds_path, listen_buffer_size);
+    LOG_IF(!m_broker_socket_uds, FATAL) << "Failed allocating memory!";
```

Comment:
LOG_IF(FATAL) is dangerous - you assume it will assert, but this depends on the easylogging configuration. We should double-check that we are always enabling LOG_IF(FATAL) to assert if we're using it like that.

Reply:
Theoretically you're right, but I don't think that in the existing logging configuration there's an option to disable FATAL assertions. It's a build flag that we don't allow controlling externally.